lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -467,12 +467,27 @@ sub prepare_job_results {
 
     # prefetch the number of available labels for those jobs
     my $job_labels = $self->_job_labels($jobs);
-    my @job_names  = map { $_->TEST } @$jobs;
+    my @job_names
+      = map { {job_template_name => $_->TEST, test_suite_name => $_->settings_hash->{TEST_SUITE_NAME} || $_->TEST} }
```

Comment:
This might slow down the page loading time notably. See my suggestion in Rocket Chat.

Reply:
I've just did a local test with OSD data. For a SLES12 build any only group "Functional: Server" the loading time increases from 0.9 seconds to 1.3 seconds. When showing the overview for the full build the loading time creases from 6.8 seconds to 11 seconds. So the performance impact is notable although it could be worse. There is likely still room for improvement.

Reply:
@Martchus I have modified the code. Is this way your suggestion in Rocket Chat? I did the performance test, but I did not found the web page that spent 11 seconds. I did test using the url such as https://openqa.suse.de/tests/overview?distri=sle&version=12-SP5&build=0366&groupid=139

Reply:
I did the test in my local and using OSD data. the performance is not good. It's almost equal to the last version. 
![1](https://user-images.githubusercontent.com/11713397/70044793-f6381080-15fd-11ea-9819-c1d72318db75.png)

Reply:
The latest version is better in the regard that it doesn't query all job settings for a job. However, it still does individual queries per job and it might be possible to retrieve all settings for the relevant jobs in one query.

And now you're also querying the test suites individually for each job which looks counterproductive.

But I needed to play around with it myself to come up with better performing code.

Reply:
@Martchus I modified the code to query the test suite one time. I did the performance test, it's better than the previous version. I do not mind that you take over the performance improvement, after all, I have spent a lot of time to do it and the performance is not good.  Thanks for your help.

Reply:
I can play around with the query a little bit based on your current state.